.circleci/config.yml

```diff
@@ -1,37 +1,319 @@
+var_7: &yarn_install
+  run: yarn install --frozen-lockfile --non-interactive
+
+# Sets up a different Docker image that includes a moe recent Firefox version which
+# is needed for headless testing.
+var_8: &docker-firefox-image
+  # TODO(devversion): Temporarily use a image that includes Firefox 62 because the
+  # ngcontainer image does include an old Firefox version that does not support headless.
+  # See the PR that fixes this: https://github.com/angular/angular/pull/26435
+  - image: circleci/node:10.12-browsers
+
+# Attaches the release output which has been stored in the workspace to the current job.
+# https://circleci.com/docs/2.0/workflows/#using-workspaces-to-share-data-among-jobs
+var_9: &attach_release_output
+  attach_workspace:
+    at: dist/releases
+
+# -----------------------------
+# Container version of CircleCI
+# -----------------------------
 version: 2
+
+# -----------------------------------------------------------------------------------------
+# Job definitions. Jobs which are defined just here, will not run automatically. Each job
+# must be part of a workflow definition in order to run for PRs and push builds.
+# -----------------------------------------------------------------------------------------
 jobs:
+
+  # -----------------------------------
+  # Build job
+  # -----------------------------------
   build:
-    docker:
-      # specify the version you desire here
-      - image: circleci/node:8.9
+    <<: *job_defaults
+    resource_class: xlarge
+    steps:
+      - *checkout_code
+      - *restore_cache
+      - *yarn_install
+
+      - run: yarn gulp :publish:build-releases
+
+      # Note: We want to save the cache in this job because the workspace cache also
+      # includes the build cache that will be updated in this job.
+      - *save_cache
+
+  # ------------------------------------------------------------------------------------------
+  # Job that runs the unit tests on locally installed browsers (Chrome and Firefox headless).
+  # The available browsers are installed through the angular/ngcontainer Docker image.
+  # ------------------------------------------------------------------------------------------
+  tests_local_browsers:
+    docker: *docker-firefox-image
+    resource_class: xlarge
+    steps:
+      - *checkout_code
+      - *restore_cache
+      - *yarn_install
+
+      - run: ./scripts/circleci/run-local-browser-tests.sh
+
+  # ----------------------------------------------------------------------------
+  # Job that runs the unit tests on Browserstack. The browsers that will be used
+  # to run the unit tests on Browserstack are set in: test/browser-providers.js
+  # ----------------------------------------------------------------------------
+  tests_browserstack:
+    <<: *job_defaults
+    resource_class: xlarge
+    environment:
+      BROWSER_STACK_USERNAME: "angularteam1"
+      BROWSER_STACK_ACCESS_KEY: "CaXMeMHD9pr5PHg8N7Jq"
```

Comment:
We should at some point figure out what to do about licenses.. I'm afraid that we all often run into rate limit exceptions and adding more projects might not improve that situation.

Reply:
We have our own, and I'll add it here to avoid using up the main one, but I just wanted to validate the format first

Reply:
Cool, good to know!